pkg/network/ebpf/c/sock.h

```diff
@@ -41,50 +42,146 @@ static __always_inline struct inet_sock *inet_sk(const struct sock *sk)
 #define fl6_sport uli.ports.sport
 #define fl6_dport uli.ports.dport
 
-#elif defined(COMPILE_RUNTIME)
+#elif defined(COMPILE_RUNTIME) || defined(COMPILE_PREBUILT)
 
+#include <linux/socket.h>
 #include <linux/tcp.h>
 #include <net/inet_sock.h>
 
 #endif // COMPILE_CORE
 
+#ifdef COMPILE_PREBUILT
+static __always_inline u64 offset_socket_sk();
+#endif
+
+static __always_inline struct sock * socket_sk(struct socket *sock) {
+    struct sock * sk = NULL;
+#ifdef COMPILE_PREBUILT
+    if (bpf_probe_read_kernel_with_telemetry(&sk, sizeof(sk), ((char*)sock) + offset_socket_sk()) < 0) {
+        return NULL;
+    }
+#elif defined(COMPILE_CORE) || defined(COMPILE_RUNTIME)
+    BPF_CORE_READ_INTO(&sk, sock, sk);
+#endif
+    return sk;
+}
+
 static __always_inline void get_tcp_segment_counts(struct sock* skp, __u32* packets_in, __u32* packets_out) {
+    if (!packets_in || !packets_out) {
+        return;
+    }
```

Comment:
Why did this get added?

Reply:
I don't think it has any affect. I think I was debugging something and thought this might be the cause. But no harm keeping it there.

Reply:
Without knowing more I'd prefer to remove it. Having code that we expect to never get hit makes reading the code harder. 

Also: from a more functional perspective, doesn't this let the programs that pass null into get_tcp_segment_counts() without triggering the verifier?   This change might make it easier to make a coding 
mistake elsewhere.